src/lib/query-parser-hoc.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import queryString from 'query-string';
+import {connect} from 'react-redux';
+
+import {detectTutorialId} from './tutorial-from-url';
+
+import {activateDeck} from '../reducers/cards';
+import {openTipsLibrary} from '../reducers/modals';
+
+/* Higher Order Component to get parameters from the URL query string and initialize redux state
+ * @param {React.Component} WrappedComponent: component to render
+ * @returns {React.Component} component with query parsing behavior
+ */
+const QueryParserHOC = function (WrappedComponent) {
+    class QueryParserComponent extends React.Component {
+        constructor (props) {
+            super(props);
+            const queryParams = queryString.parse(location.search);
+            this.state = {
+                tutorial: false
+            };
+
+            const tutorialId = detectTutorialId(queryParams);
+            if (tutorialId) {
+                this.state.tutorial = true;
+                if (tutorialId === 'all') {
+                    this.openTutorials();
+                } else {
+                    this.setActiveCards(tutorialId);
+                }
+            }
+        }
+        setActiveCards (tutorialId) {
+            this.props.onUpdateReduxDeck(tutorialId);
+        }
+        openTutorials () {
+            this.props.onOpenTipsLibrary();
+        }
+        render () {
+            const {
+                hideIntro: hideIntroProp,
```

Comment:
I think you should list onOpenTipsLibrary and onUpdateReduxDeck here, so they get excluded from componentProps (and don't cause console errors)